### ISSUE TITLE
Issues with reCAPTCHA v2 inside a modal in Internet Explorer 8 and up

### ISSUE THREAD

**Issue by emilhem** [twbs/bootstrap#16037]
I use reCAPTCHA v2 on a website using the `grecaptcha.render` function.

I have tested it in Firefox and Chrome without any issues.

In Internet Explorer 8 - 11 it won't work. I've tested it on multiple computers and different networks.

It won't allow a correct entry of the captcha nor will it allow for it's submission when inside a Bootstrap Modal.

YouTube video to display the issue: https://youtu.be/KnaQqqaz-00

JSFiddle to demonstrate the issue: http://fiddle.jshell.net/emilhem/Lgovn28f/4/

Does anyone else experience this issue? Is there a solution to the issue I'm experiencing?

**Comment by twbs-lmvtfy** [comment]
Hi @emilhem!

You appear to have posted a live example (https://fiddle.jshell.net/emilhem/Lgovn28f/2/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 15, column 98: ```` & ```` did not start a character reference. (```` & ```` probably should have been escaped as ```` &amp; ````.)

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by emilhem** [comment]
Fixed JSFiddle: http://fiddle.jshell.net/emilhem/Lgovn28f/4/

Extremely picky HTML5 validator...

**Comment by cvrebert** [comment]
Sorry, but we are unable to provide assistance for issues involving third-party widgets/libraries, such as reCAPTCHA. See also http://getbootstrap.com/javascript/#callout-third-party-libs
Please try asking on StackOverflow (or similar) instead. Thanks!

**Comment by cvrebert** [comment]
Although your use of `setTimeout` instead of `.on('shown.bs.modal', ...)` can't be helping things.

**Comment by marcvangend** [comment]
I am honestly amazed that this is how Bootstrap treats a contributor to the project. Here is someone who has spent time and effort to provide a useful, constructive bug report, only to be smacked in the face, first by a bot, then by a wont-fix from a maintainer who didn't investigate the cause of the problem.

It may or may not be a coincidence, but on the exact same day that this issue was opened and closed, a 2 months old [question on StackOverflow](http://stackoverflow.com/q/27886618/241546) received an [answer](http://stackoverflow.com/a/28965638/241546) containing a workaround which looks like this:

```
$.fn.modal.Constructor.prototype.enforceFocus = function () { };
```

In other words, the problem is in a function coming from Bootstrap itself. It probably serves a purpose, but breaks other scripts by stealing focus. Especially since this "other script" is not some hackish piece of custom JS, but Google's Recaptcha, I kindly ask you to reconsider your decision and offer @emilhem your apologies.

**Comment by atomgregg** [comment]
How is this still closed?

It's been kindly pointed out already that the problem is with a function that by all means seems to be headed towards the deprecated pile in v4.

Any chance of fixing the bug and supporting those still using the official release version and not yet on the alpha train?

**Comment by bardiharborow** [comment]
Bootstrap receives a large quantity of low quality bug reports where the user is at fault rather than Bootstrap. This may not be the case with your bug report, but we can't always determine this at first glance and due to the number of issues we deal with we often default to closing. Getting angry at the core team (who are just volunteers doing their best), is not helpful.

@marcvangend seems to have demonstrated that it's a problem on our end, so I've added this issue to my review list. I can't however make any guarantees about backporting a fix as [support for v3 has mostly ceased](https://github.com/twbs/bootstrap/issues/20631).
